code/game/objects/explosion_iterative.dm

```diff
@@ -0,0 +1,436 @@
+
+
+
+
+/obj/effect/debugging/marker/var/changedtimes = 0
+/obj/effect/debugging/marker/var/powerexplosion = 0
+/turf
+	var/explosion_ratio = 1.0 //this affect the explosion spread, but by a % instead of direct value (should make walls a bit more meaningfull vs stronger bombs
+	var/datum/side_explosion/side_explosion_data = 0
+
+/turf/space
+	explosion_resistance = 10
+
+/turf/simulated/floor
+	explosion_resistance = 1
+
+/turf/simulated/mineral
+	explosion_resistance = 2
+
+/turf/simulated/shuttle/floor
+	explosion_resistance = 1
+
+/turf/simulated/shuttle/floor4
+	explosion_resistance = 1
+
+/turf/simulated/shuttle/plating
+	explosion_resistance = 1
+
+/turf/simulated/shuttle/wall
+	explosion_resistance = 5
+	explosion_ratio = 0.80
+/turf/simulated/wall
+	explosion_resistance = 5
+	explosion_ratio = 0.80
+
+/turf/simulated/wall/r_wall
+	explosion_resistance = 15
+	explosion_ratio = 0.60
+
+//Code-wise, a safe value for power is something up to ~25 or ~30.. This does quite a bit of damage to the station.
+//direction is the direction that the spread took to come to this tile. So it is pointing in the main blast direction - meaning where this tile should spread most of it's force.
+
+/proc/dirtoangle(direction) //convert to angle starting from north
+{
+	if (direction == NORTH)
+		return 0
+	if (direction == SOUTH)
+		return 180
+	if (direction == EAST)
+		return 90
+	if (direction == WEST)
+		return 270
+}
+/proc/highPrioritySleep(time, debugname) //just a placeholder until i reimplement my process tracking system
+	sleep(time)
+
+/obj/effect/fakefire
+	anchored = 1
+	mouse_opacity = 0
+	unacidable = 1//So you can't melt fire with acid.
+	icon = 'icons/effects/fire.dmi'
+	icon_state = "1"
+	layer = TURF_LAYER
+	blend_mode = BLEND_ADD
+	New()
+		color = heat2color(25000)
+		set_light(color)
+
+datum/line_explosion_controller //used to manage the explosion itself
+	var/highestsidepower=0 //this is a value used to not have to iterate everytime through all the queued side. We only iterate to pick the next start point
+	var/list/queue_explosion_lines = list() //list of lines that we are currently exploding
+	var/list/queue_explosion_sides = list() //list of side explosions that we use to start new lines from. We find the highest power side explosion end start a line for each of the same power
+	var/turf/epicenter = 0
+	var/power = 0
+	var/list/explosion_turfs //list containing our calculated explosion to apply
+	var/delayBetweenWave = 1 //how long we sleep between each explosions wave
+	var/iterationperWave = 7 //how many steps between sleeps
+	var/destructionperWave = 15 //how many tiles do we apply the explosion on between sleeps
+	var/calculating = 0
+	var/devastation_boost_count = 0
+	var/heavy_impact_boost_count = 0
+	var/light_impact_boost_count = 0
+	var/highpower = 0
+	var/midpower = 0
+	var/lowpower = 0
+	proc/startExplosion(turf/m_epicenter, devastation_range, heavy_impact_range, light_impact_range)
+	{
```

Comment:
on the curly bracket its just an old habbit. 

as for the relative path, is simply because i find it cleaner for variable declaration in an object. I can change it if you insist i dont really care for it that much either way